JavaScript/JavaScriptSDK/appInsights.ts

```diff
@@ -185,56 +186,56 @@ module Microsoft.ApplicationInsights {
-                                    +pageViewPerformance.duration :
-                                    durationMs,
-                                properties,
-                                measurements);
-
-                            if (pageViewPerformance.isValid) {
-                                var pageViewPerformanceData = new ApplicationInsights.Telemetry.Common.Data<ApplicationInsights.Telemetry.PageViewPerformance>(
-                                    Telemetry.PageViewPerformance.dataType, pageViewPerformance);
-                                var pageViewPerformanceEnvelope = new Telemetry.Common.Envelope(pageViewPerformanceData, Telemetry.PageViewPerformance.envelopeType);
-                                this.context.track(pageViewPerformanceEnvelope);
+                return;
+            }
+
+            var start = Telemetry.PageViewPerformance.getPerformanceTiming().navigationStart;
+
+            if (this.config.relativePageViewDuration) {
+                var duration = Telemetry.PageViewPerformance.getDuration(start, +new Date);
+                this.sendPageViewInternal(name, url, duration, properties, measurements);
+                this.flush();
+            }
+
+            var maxDurationLimit = 60000;
+            var handle = setInterval(() => {
+                try {
+                    if (Telemetry.PageViewPerformance.isPerformanceTimingDataReady()) {
+                        clearInterval(handle);
+                        var pageViewPerformance = new Telemetry.PageViewPerformance(name, url, null, properties, measurements);
+                        
+                        if (pageViewPerformance.getIsValid()) {
+                            if (!this.config.relativePageViewDuration) {
+                                this.sendPageViewInternal(name, url, pageViewPerformance.getDurationMs(), properties, measurements);
                             }
 
-                            this.context._sender.triggerSend();
+                            var pageViewPerformanceData = new ApplicationInsights.Telemetry.Common.Data<ApplicationInsights.Telemetry.PageViewPerformance>(
+                                Telemetry.PageViewPerformance.dataType, pageViewPerformance);
+                            var pageViewPerformanceEnvelope = new Telemetry.Common.Envelope(pageViewPerformanceData, Telemetry.PageViewPerformance.envelopeType);
+                            this.context.track(pageViewPerformanceEnvelope);
                         }
-                    } catch (e) {
-                        _InternalLogging.throwInternalNonUserActionable(LoggingSeverity.CRITICAL, "trackPageView failed on page load calculation: " + Util.dump(e));
+
+                        this.flush();
                     }
-                }, 100);
-            }
+                    else if (Telemetry.PageViewPerformance.getDuration(start, +new Date) > maxDurationLimit) {
+                        clearInterval(handle);
```

Comment:
Because maxDurationLimit reached and performance timing data is not available yet (see previous "if"). I.e. we don't have perf data to report and we don't want to wait any longer, and we still want to report page view.